crates/proto/src/xfer/dns_multiplexer.rs

```diff
@@ -17,51 +17,50 @@ use std::sync::Arc;
 use std::task::{Context, Poll};
 use std::time::{Duration, SystemTime, UNIX_EPOCH};
 
-use futures_channel::oneshot;
+use futures_channel::mpsc;
 use futures_util::stream::{Stream, StreamExt};
 use futures_util::{future::Future, ready, FutureExt};
 use log::{debug, warn};
 use rand;
 use rand::distributions::{Distribution, Standard};
-use smallvec::SmallVec;
 
 use crate::error::*;
 use crate::op::{Message, MessageFinalizer, OpCode};
 use crate::xfer::{
     ignore_send, DnsClientStream, DnsRequest, DnsRequestOptions, DnsRequestSender, DnsResponse,
-    DnsResponseFuture, SerialMessage, CHANNEL_BUFFER_SIZE,
+    DnsResponseStream, SerialMessage, CHANNEL_BUFFER_SIZE,
 };
 use crate::DnsStreamHandle;
 use crate::Time;
 
 const QOS_MAX_RECEIVE_MSGS: usize = 100; // max number of messages to receive from the UDP socket
 
 struct ActiveRequest {
-    // the completion is the channel for a response to the original request
-    completion: oneshot::Sender<Result<DnsResponse, ProtoError>>,
-    request_id: u16,
-    request_options: DnsRequestOptions,
+    // the completion is the channel for a response to the original request.
     // most requests pass a single Message response directly through to the completion
     //  this small vec will have no allocations, unless the requests is a DNS-SD request
-    //  expecting more than one response
-    // TODO: change the completion above to a Stream, and don't hold messages...
-    responses: SmallVec<[Message; 1]>,
+    //  expecting more than one response.
+    //  FIXME: change to a Bounded channel
```

Comment:
I think that should probably be fixed as part of this PR.

Reply:
This is a draft. I was just putting it up for early feedback as I make progress. There are more changes coming, this being one of them. It will end up causing a big ripple effect which is why I marked it with the FIXME.

Reply:
Sure, I just wanted to make sure this doesn't end up as a FIXME on main.

Reply:
I was thinking of adding a cleanliness pass for FIXMEs. I tend to use them has “must be fixed before merge”, and TODOs are treated more as “should ideally have an associated issue, but doesn’t block merging”